### PR TITLE
Bluetooth: controller: legacy: Ensure that a `case` statement is present

### PR DESCRIPTION
**Pull request by cvinayak**
When compiling a sample like a beacon, none of the `if defined()`
statements in the switch case are actually compiled in, so the compiler
complains of an orphan mem_release();break; sequence.
Add a type that we know will always be compiled in in order to avoid
this issue.

Relates to commit 76bfea7cf90d ("Bluetooth: controller: Ensure that a
`case` statement is present").

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>
Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Review by thoh-ot — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Comment by cvinayak** [comment]
@mbolivar do approve if the cherry-pick of this helped you pass CI...

@carlescufi and @aescolar you can merged once @mbolivar approves this PR

**Pull request merged**
